stripe-ui-core/src/main/java/com/stripe/android/uicore/analytics/UiEventReporterCompose.kt

```diff
@@ -6,24 +6,33 @@ import androidx.compose.foundation.interaction.MutableInteractionSource
 import androidx.compose.foundation.interaction.PressInteraction
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
-import androidx.compose.runtime.compositionLocalOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.saveable.Saver
 import androidx.compose.runtime.saveable.SaverScope
 import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.staticCompositionLocalOf
 import com.stripe.android.core.Logger
 import com.stripe.android.uicore.BuildConfig
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.parcelize.Parcelize
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-val LocalUiEventReporter = compositionLocalOf<UiEventReporter> {
+val LocalUiEventReporter = staticCompositionLocalOf<UiEventReporter> {
     EmptyUiEventReporter
 }
 
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class InteractionReporterElements(
+    val mutableInteractionSource: MutableInteractionSource,
+    val reportInteractionManually: () -> Unit
+) {
+    operator fun component1() = mutableInteractionSource
```

Comment:
This is the magic at the callsites that I think makes it far less easy to read, we should avoid this!